digital_land/collection.py

```diff
@@ -555,3 +557,45 @@ def entry_date(self, entry):
         return entry.get(
             "entry-date", datetime.utcnow().strftime("%Y-%m-%dT%H:%H:%M:%SZ")
         )
+
+    def retire_endpoints_and_sources(collection, collection_df_to_retire) -> None:
+        """
+        This method should be called from commands.py with the correct input.csv format per unique collection.
+
+        Args:
+            collection (Collection): The Collection object which gets instantiated by a corresponding commands.py function which calls this method.
+            collection_df_to_retire (str): A small dataframe of sources and endpoints to be retired within a colleciton, this should be
+                generated by a function call within commands.py.
+        """
+
+        try:
+            endpoint_csv_path = os.path.join(collection.dir, "endpoint.csv")
+            source_csv_path = os.path.join(collection.dir, "source.csv")
+
+            # Read endpoint and source CSV files
+            endpoint_csv_df = pd.read_csv(endpoint_csv_path)
```

Comment:
Consider adding a check to ensure the CSV files exist before attempting to read them

Reply:
The try/catch checks for FileNotFoundErrors so we should be okay